### ISSUE TITLE
License issues

### ISSUE THREAD

**Issue by Urisma** [piorrro33/psxjin#64]
There's a slew of them. 

The MIT license is still left over from this commit: https://github.com/piorrro33/psxjin/commit/2672fdd9b41ff654f64df32df5b17cf4a877b841

This commit removes a ton of headers from files licensed under various licenses: https://github.com/piorrro33/psxjin/commit/3b4503139d15cb6030aeee50e13f6ba0e4487269

It even removes headers that specifically say they can't be removed and must accompany the code.

For some stackoverflow reference: http://stackoverflow.com/questions/14762576/changing-copyright-notice-in-file-header-when-contributing-to-a-project

**Comment by piorrro33** [comment]
Well, you're right. We'll have a ton of work to do, but thanks for remembering us the law. :smiley: 
@vgturtle127 Looks like you did something wrong during your cleanup.

**Comment by piorrro33** [comment]
Latest commit sets back the GPLv3 license : https://github.com/piorrro33/psxjin/commit/eb21f83ba689dc3b19019f1bf9e191b7b378fe92
For the headers, @vgturtle127 will do the work, please.

**Comment by vgturtle127** [comment]
Let me get permission from the Desmume and PCSX2 teams, and I will ask the PSXjin team.

**Comment by Urisma** [comment]
I'm just going to reply here to what was said both in this thread and on comment replies because this issue is much more visible. I've quoted the relevant parts.

>I wish the GNU GPL had some sort of exception or clause about this. I know they are trying to prevent people from switching, but it is really difficult when you just want to switch to a similar license even.

That's the entire point. That the license can't be changed without someone's express permission. Take a look at GCC relicensing from v2 to v3 and how the different BSDs reacted.

>I'm not going to put the license headers back in, as I have said before it is redundant to have all the files with license information in them. Not only that, but some of them were wrong outright anyway.

It's redundant because it's trying to prevent exactly what you are doing. Namely stealing code. Which is what you've done by removing license headers. Let's take a look at some examples from your previous commit. [This header](http://pastebin.com/VkJnGAVx) was removed from a file that belongs to the desmume project. You removed all the copyright statements. This is theft. This is not legal. [Here](http://pastebin.com/BpfyWjd9) is another one. By removing the header you are stealing from the copyright holders, William D. Herndon and Open Foundation Classes. Andrea Mazzoleni is another person you've stolen from. And Olli Parviainen. And Pete Bernert, who, if you'd have read the license header, specifically states that his work is GPL v2+. and Darko Matesic. And lots of others that I can't find because that commit is so hilariously large. [Take a look at this one](http://pastebin.com/rgrkZJac) it *plainly* says that the header must accompany the code. And you've removed it.

So I'll be reporting this repo to GNU for GPL violations. I'll report it to Microsoft as well as any other copyright holders if I find the chance.

And also, I submitted a few changes to PCSX2. You haven't stolen them yet, but I'm going to go ahead and state that I give absolutely no permission for my changes to PCSX2 to be relicensed from GPL v2.

**Comment by vgturtle127** [comment]
So I tried to make what I did right again, and you are making it so I can't use any PCSX2 code at all? I wasn't trying to make anybody angry.

And I'm stealing code? Hm.

I got permission from most of the people on some of the teams already...

I can revert the changes back, but I will still be breaking the PCSX2 part if none of your parts are able to be licensed. So I guess we will have to start from scratch?

If we can't use PCSX2 that means we can't use any of our GPU code, SPU code, or any of our patches in the recompiler or interpreter.

I guess we will have to clear the repo and start over @piorrro33 . :disappointed:

**Comment by Urisma** [comment]
I'm saying you can't use my changes to PCSX2.

Yes you are stealing code by removing license headers.

All you have to do is put them back.

**Comment by vgturtle127** [comment]
But we would still be using PCSX2 code, which we no longer have permission to use.

We will just have to reset everything, it would be too much work to try and only put the headers back AND remove the infringing stuff.

**Comment by Urisma** [comment]
No. You only don't have permission to use what I've submitted to PCSX2. Which doesn't even exist in this repo. You mentioned PCSX2 so I wanted to be explicit that my changes (a whopping 50 lines or so) can't be relicensed if you decide to pull anything new from PCSX2.

Literally all you need to do is add back the 50 or so license headers you removed.

**Comment by vgturtle127** [comment]
But if one person doesn't allow us to use any part of the code, we can't use any of it. That's that getting permission part that sucks. If one person says no, that means everybody says no.

I was told by a PCSX2 member we should speak to somebody at GNU, and in the meantime we should clear the repo to make sure none of us get any legal action. None of us have any money, since we are all hobbyist open-source programmers, and anybody who sued us would instantly win because we can't afford lawyers. But we wouldn't be able to pay for anything, so that means jail time. None of us want that.

**Comment by Urisma** [comment]
> But if one person doesn't allow us to use any part of the code, we can't use any of it. That's that getting permission part that sucks. If one person says no, that means everybody says no.

No... You can use the code for whatever as long as it adheres to the original license. I have literally zero code under my own copyright in this repo. You can still use all the code from PCSX you want. The only thing I have power over is my 50 lines in the PCSX2 repo, and the only power I have is to say that someone else can't relicense them. You can still bring them into the repo and use them all you want.

Literally all you need to do is add back in the original headers. That will solve all the legal problems and you can go back to writing code.

**Comment by vgturtle127** [comment]
But bringing back the headers doesn't clear up all the issues, as you say. We would still have to ask everybody for permission, and on top of that we would have to be able to contact all the individuals responsible.

We can't use PCSX2 if anybody disagrees with our use of the code. It wouldn't hold up legally.

**Comment by Urisma** [comment]
For permission to relicense. That's it. If you stick with GPL v2 you can do whatever you want with the code as long as you keep it GPL v2. You can use it, fork it, and rewrite it to your heart's content. Just don't relicense it. That's literally what the headers are saying.

**Comment by vgturtle127** [comment]
We never said we wanted to relicense everything. We only wanted to remove all the useless comments in the files. I'm not sure how having a license in the repo which clearly states GPLv2 would be relicensing anything. If the license is in the code and we move it to a files, what is the difference?

This is why I hate forking projects. :-1:

**Comment by Urisma** [comment]
Removing the copyright notices is theft. You can't do that. You might be able to relocate the licenses and copyright notices to a specific authors file, but I'm not sure. The license itself should probably give more explanation.

 You just have to keep the headers and the copyright notices. That's literally all you need to do and everything is fine. Literally everything will be legal and ok again if you literally just add back the copyright notices and headers. I will even give you permission to relicense my 50 lines in PCSX2 to GPL v2 if you just add back the license headers that you removed. That doesn't matter since you aren't planning on relicensing and aren't using my changes, but you seem fixated on that so why not.

**Comment by piorrro33** [comment]
If they had to ask every person that contributed to their project, how could Dolphin relicense to GPL v3? I really don't know. Though, we HAVE to put back these license headers for now on, and THEN ask them if they are ok with us doing anything with their code. I think that's the best idea because even if they don't want to give us their code, we'll still be able to use the code without any problem, just by keeping the headers.

**Comment by vgturtle127** [comment]
Dolphin only got permission from most of the team. They had talked to a lawyer and apparently they only needed 90% of the teams permission to proceed with the re-licensing. And that was from GPLv2 to GPLv2+.

I am putting the license headers and stuff back on, but we will have to do all the changes again, so it may take me awhile.

I will create a AUTHORS.md and CONTRIBUTING.md file to help minimize the blow, but I'm guessing that the FSF and the other emulator teams aren't going to give us permission based on @Urisma and also the immense amount of hateful comments from most people.

**Comment by piorrro33** [comment]
"the immense amount of hateful comments" Source?

**Comment by Urisma** [comment]
I don't know how to make this any clearer.

You do NOT have to ask any authors if you can modify their source.

This means you don't have to ask the Desmume team or PCSX team or whatever if you can incorporate their code or change it.

You only have to ask for permission from original authors if you are planning to relicense the project. You aren't doing that. Therefore, you can use everyone's code with impunity, given that you dont remove the headers.

That is literally it. You do not need permission from anyone to do anything you are planning to do with their code, unless you are relicensing. You can delete it, make it better, write about how god awful it is (a whole lot of the code is pretty crappy), whatever.

You just can't remove the license headers.

That is it.

Put the license headers back. That's all I am asking. That will make EVERYTHING ok. There will be no legal issues if you just PUT THE HEADERS BACK.

I don't know how to make this any more clear.

Put the headers back.

**Comment by Urisma** [comment]
And if by "immense amount of hateful comments from most people", you mean [this](https://www.reddit.com/r/emulation/comments/3dleso/psxjin_development_has_started_back_up_now_on/) post to /r/emulation and the comment that I made, tough. People are going to hate on you when you steal code. Make it right by PUTTING THE HEADERS BACK and everything will be ok. I'll even post a followup about how nice it was that y'all responded swiftly after this was all brought to your attention.

**Comment by Urisma** [comment]
Well boys I went ahead and decided to fix it myself. See https://github.com/piorrro33/psxjin/pull/65

**Comment by vgturtle127** [comment]
I was going to say something about how I didn't steal code since I was going to externalize the entirety of what was removed into a file, but I doubt anybody cares who is involved (or not) really.

You don't have to do that, our reputation has already been somewhat tattered. I think it would be better if we just left it alone. If you try to say we did something good, they will probably just say something along the lines of "just you wait"...
